app/javascript/article-form/components/ArticleCoverImage.jsx

```diff
@@ -53,55 +58,75 @@ export class ArticleCoverImage extends Component {
     });
   };
 
+  onDropImage = (event) => {
+    onDragExit(event);
+
+    if (event.dataTransfer.files.length > 1) {
+      addSnackbarItem({
+        message: 'Only one image can be dropped at a time.',
+        addCloseButton: true,
+      });
+      return;
+    }
+
+    this.handleMainImageUpload(event);
+  };
+
   render() {
```

Comment:
Are we worried that the render now goes over the 50 lines limit?

Reply:
Not too much honestly. The line bust is because of prettier formatting. It could be broken up a bit though. I'll take a peek.

Reply:
I took a peek and there some other refactorings that could go on in there, but I would probably leave it for a subsequent PR.